.github/workflows/code-quality-tests.yaml

```diff
@@ -0,0 +1,53 @@
+name: Code Quality Tests
+
+on:
+  push:
+    branches:
+      - '**'
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+      - 'mkdocs.yml'
+      - 'requirements.txt'
```

Comment:
if you change or remove some dependencies could it break the tests ?

Reply:
It could only break pre-commit and linter tests as thats the only for docs and we are going to run those tests.
Except MKdocs the repo does not use these packages in anyway.